### DISCUSSION TITLE
Docker and file permissions

### DISCUSSION THREAD

**Discussion by robflate** [garethgeorge/backrest#471] [Q&A]
I'm running Backrest in Docker. When it creates a repo and subsequent snapshots, all file permissions of the original files are ignored and all files are owned by root when restored. How can I run Backrest in Docker and respect file permissions and `user:group`?

Also, how can I run the container as a non-root user but still run `HOOK` commands with `docker` e.g. `docker stop container`? If I set `user: 1000:1000` (my host user) in my compose, I get permission denied errors when running docker commands. Incidentally, I can run rclone commands. Also, when using `user: 1000:1000`, Backrest fails on files where that user doesn't have permission inside the container.

To simplify, I want to;

- Run the container as `1000:1000` so bind mounts (`/data`, `/config`, `/repos` etc) have the same owner as my host.
- Respect file permissions when restoring files.
- Run docker commands without getting permission denied.
- Run Backups on files/folders regardless of `user:group`.

Maybe it would be possible to to have env vars `PUID` and `PGID` that set the user inside the container. This user could also be added to the `Docker` group inside the container?

Thanks.

**Comment by garethgeorge** [comment]
It looks like it's common for a container to have an entrypoint.sh script that includes something like 


```
PUID=${PUID:-1000}
PGID=${PGID:-1000}

USER=abc
GROUP=abc

if [ "$PUID" == "" -o "$PUID" == "0" ]; then
	USER=root
	GROUP=root
elif ! $(id abc&>/dev/null); then
	groupadd -g $PGID $USER
	useradd -m -u $PUID -g $GROUP $USER
fi
```

It's reasonable that backrest should support this :) please open a feature request for this.

**Reply by janstadt** [reply]
Did this get implemented? i get this when clicking Check Now on a repository:

```
[unknown] auto unlock repo "Seed": command "/bin/nice -n -10 ionice -c 2 -n 0 /bin/restic unlock -o sftp.args=-oBatchMode=yes" failed: exit status 1
Output:
nice: setpriority(-10): Permission denied
```

The mounted volumes are owned by root which is how the container is running.

**Reply by garethgeorge** [reply]
Hey, your container must be running as privileged to use a higher than default niceness.
